projects/telemetry/src/nmea_checksum.c

```diff
@@ -0,0 +1,82 @@
+#include "nmea_checksum.h"
+#include <stdbool.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+
+#define NMEA_CHECKSUM_HEX_ERROR 16
+
+// Private method to convert hex char to int
+static uint8_t prv_hex_to_int(char h) {
+  if ('0' <= h && h <= '9') {
+    return h - '0';
+  } else if ('A' <= h && h <= 'F') {
+    // We only care about uppercase because the NMEA messages are all caps
+    return h - 'A' + 10;
+  }
+  // Just to make the checksum fail
+  return NMEA_CHECKSUM_HEX_ERROR;
+}
+
+static StatusCode prv_checksum_to_int(char tens, char ones, uint8_t *computed) {
+  uint8_t int_tens = prv_hex_to_int(tens);
+  uint8_t int_ones = prv_hex_to_int(ones);
+
+  // Checks for the 16 because it is the result sent in case of error
+  if (ones == NMEA_CHECKSUM_HEX_ERROR || tens == NMEA_CHECKSUM_HEX_ERROR) {
+    return status_msg(STATUS_CODE_INVALID_ARGS, "Could not convert hex chars to ints\n");
+  }
+  if (computed != NULL) {
+    // Bitwise operation is equivalent to "int_tens * 16 + int_ones"
+    *computed = ((int_tens & 0xF) << 4) | (int_ones & 0xF);
```

Comment:
Just FYI: The compiler probably would've optimized it into the same code.

Your comment isn't actually fully accurate, as the bitwise approach is more "correct" since it masks the values. It's only because you always limit both variables to `[0x00, 0x0F]` that they're equivalent, and if you trust that will always be the case, you can actually remove the masks. Just something to keep in mind.